provisioner/provisioner.go

```diff
@@ -598,3 +600,37 @@ func (p *Provisioner) executeCnspec(ui packer.Ui, comm packer.Communicator) erro
 
 	return nil
 }
+
+func updateOsProvider(ui packer.Ui) {
+	allProviders, err := providers.ListActive()
+	if err != nil {
+		ui.Error(err.Error())
+		ui.Message("failed to list providers, not going to update cnspec os provider")
+		return
+	}
+	outdated := false
+	for _, provider := range allProviders {
+		if provider.Name == "os" {
```

Comment:
is it just the os provider that we are interested in? is it possible that other providers are also used by the packer plugin? We do have types in OS that are defined in network, for example... Shouldn't this be just the same function as the one updating the providers in cnspec?

Reply:
The network provider is a good catch. Didn't think about this one. Yes, then we can use the same code as for the cnspec PR.